### PR TITLE
674 update workspace

### PR DESCRIPTION
**Pull request by Dhruv-Copods**
Feature:
Created Edit Workspace Name Feature

Had to create a Query call to update the Workspace

Created an Interface for Settings Workspace Object

Had to modify the Header component in order to make it dynamic in relation to size.

Created updatedById Key in Workspace Schema which will have an empty string as the default value and will update only on updating workspace. Also had to create a migration for this.

Updated the Interface of the InputField Component as it is not necessary to make label as a required field. Made changes in InputField Component itself also.

### CODE REVIEW

File: app/components/settings/Workspace.tsx [hunk 1]
```diff
@@ -1,75 +1,201 @@
...
+    </div>
+  )
+}
+
+const EditButton = ({
+  setIsEdit,
+  t,
+}: {
+  setIsEdit: (val: boolean) => void
+  t: TFunction<"translation", undefined>
```
Comment (ayushicopods) [inline app/components/settings/Workspace.tsx:58]: do we need a translate function if we are importing it globally then why do we need to pass it an argument

File: cypress/support/common-function.ts [hunk 2]
```diff
@@ -196,7 +196,7 @@ export const getDeleteTest = () => cy.get(".deleteTest")
 export const getConfirmDelete = () => cy.get("#confirm-delete")
 export const getToastifyCloseBtn = () => cy.get(".Toastify__close-button")
 export const getAssesPageTitle = () => cy.get("#add-assessment-page-title")
-export const getSectionInDiv = () => cy.get("div#section")
+export const getSectionInDiv = () => cy.get("div#section", { timeout: 60000 })
```
Comment (ayushicopods) [inline cypress/support/common-function.ts:199]: why timeout of 1 minute?

File: app/components/settings/Workspace.tsx [hunk 3]
```diff
@@ -1,75 +1,201 @@
...
   const [showLeaveWorkspacePopup, setShowLeaveWorkspacePopup] = useState(false)
+  const [isEdit, setIsEdit] = useState(false)
+  const workspaceLoaderData = useLoaderData()
+  const [inputValue, setInputValue] = useState(
+    workspaceLoaderData?.userWorkspaces?.find(
+      (workspaces: SettingWorkspace) =>
+        workspaces?.workspaceId === workspaceLoaderData?.currentWorkspaceId
+    )?.workspace?.name
+  )
+  console.log(workspaceLoaderData)
```
Comment (ayushicopods) [inline app/components/settings/Workspace.tsx:84]: remove console

File: app/components/settings/Workspace.tsx [hunk 4]
```diff
@@ -1,75 +1,201 @@
...
 import Button from "../common-components/Button"
 import DialogWrapper from "../common-components/Dialog"
 
+import { WorkspaceDetailsSection } from "./WorkspaceDetailsSection"
+
+import Header from "~/components/header/Header"
+import type { SettingWorkspace } from "~/interface/Interface"
+
+const CancelAndSaveButton = ({
+  inputValue,
```
Comment (ayushicopods) [inline app/components/settings/Workspace.tsx:16]: we can change the name

File: app/models/workspace.server.ts [hunk 5]
```diff
@@ -131,3 +131,15 @@ export async function getOwnersWorkspaces(userId: string) {
   })
   return ownerWorkspaceId
 }
+
+export async function updateUserWorkspace(
+  id: string,
+  name: string,
+  updatedById: string
+) {
+  console.log(id, name, updatedById)
```
Comment (ayushicopods) [inline app/models/workspace.server.ts:140]: remove console

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**